syntax/parse.go

```diff
@@ -36,13 +42,16 @@ func Parse(filename string, src interface{}) (f *File, err error) {
 	if f != nil {
 		f.Path = filename
 	}
+	if mode&RetainComments != 0 {
+		p.assignComments(f)
+	}
 	return f, nil
 }
 
 // ParseExpr parses a Skylark expression.
 // See Parse for explanation of parameters.
-func ParseExpr(filename string, src interface{}) (expr Expr, err error) {
-	in, err := newScanner(filename, src)
+func ParseExpr(filename string, src interface{}, mode Mode) (expr Expr, err error) {
+	in, err := newScanner(filename, src, mode&RetainComments != 0)
```

Comment:
Doesn't this function need a call to p.assignComments? How to comments get attached?

Reply:
Done